app/src/main/java/org/stepic/droid/core/presenters/TagListPresenter.kt

```diff
@@ -0,0 +1,131 @@
+package org.stepic.droid.core.presenters
+
+import io.reactivex.Observable
+import io.reactivex.Scheduler
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.functions.BiFunction
+import io.reactivex.subjects.PublishSubject
+import org.stepic.droid.core.presenters.contracts.CoursesView
+import org.stepic.droid.di.qualifiers.BackgroundScheduler
+import org.stepic.droid.di.qualifiers.MainScheduler
+import org.stepic.droid.di.tags.TagScope
+import org.stepic.droid.model.Course
+import org.stepic.droid.model.Meta
+import org.stepic.droid.model.Tag
+import org.stepic.droid.util.resolvers.SearchResolver
+import org.stepic.droid.web.Api
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import javax.inject.Inject
+
+@TagScope
+class TagListPresenter
+@Inject
+constructor(
+        private val tag: Tag,
+        private val api: Api,
+        private val searchResolver: SearchResolver,
+        @MainScheduler
+        private val mainScheduler: Scheduler,
+        @BackgroundScheduler
+        private val backgroundScheduler: Scheduler
+) : PresenterBase<CoursesView>() {
+    companion object {
+        private const val FIRST_PAGE = 1
+    }
+
+
+    private val currentPage = AtomicInteger(1)
+    private val hasNextPage = AtomicBoolean(true)
+    private val compositeDisposable = CompositeDisposable()
+    private val publisher = PublishSubject.create<Int>()
+
+    fun onInitTag() {
+        val disposable = publisher
```

Comment:
If we face an error this disposable will not work anymore and we have to resubscribe to Publisher or suppress that errors.
e.g. 
1. user opens `TagFragment`
2. faces connection problems due to some network instability
3. swipes down to refresh page and see endless progress bar
<img src="https://user-images.githubusercontent.com/3856523/33126733-a3154378-cf96-11e7-861e-719ec8984280.png" width="300" />

Reply:
Not it is resubscribed in onError